src/Repository/FileRepository.hpp

```diff
@@ -33,16 +34,25 @@ Copyright_License {
  * The announced contents of a remote file repository.  These are the
  * files that can be downloaded by the XCSoar user.
  */
-struct FileRepository {
+class FileRepository {
   typedef std::list<AvailableFile> FileList;
   typedef FileList::const_iterator const_iterator;
 
+private:
   FileList files;
 
+  void ParseFileRepository(Path repofile_path);
+  void AddFile(AvailableFile &file);
+
+public:
   void Clear() {
     files.clear();
   }
 
+  void Load(Path repofile_path);
+  void Download(std::string name);
+  bool IsOutOfDate(std::string name);
```

Comment:
This requires the caller to copy its name to a newly allocated `std::string`. Unnecessary bloat!